aspnetcore/fundamentals/startup.md

```diff
@@ -47,18 +47,58 @@ To learn more about `WebHostBuilder`, see the [Hosting](xref:fundamentals/hostin
+
+The following code sets the compatibility mode to ASP.NET Core 2.1:
+
+[!code-csharp[Main](startup/sampleCompatibility/Startup.cs?name=snippet1)]
+
+We recommend you test your application using the latest version (`CompatibilityVersion.Version_2_1`). We anticipate that most applications will not have breaking changes using the latest version. 
+
+Applications that call `SetCompatibilityVersion(CompatibilityVersion.Version_2_0)` are protected from potentially breaking changes introduced in the ASP.NET Core 2.1 MVC runtime and later 2.x versions. This protection:
+
+* Does not apply to all 2.1 and later changes, it's targeted to potentially breaking ASP.NET Core MVC runtime changes.
+* Does not extend to the next major version.
+
+The default compatibility for ASP.NET Core 2.1 and later 2.x applications that do **not** call `SetCompatibilityVersion` is 2.0 compatibility. That is, not calling `SetCompatibilityVersion` is the same as calling `SetCompatibilityVersion(CompatibilityVersion.Version_2_0)`.
+
+The following code sets the compatibility mode to ASP.NET Core 2.1, except for the following behaviors:
+
+* [AllowCombiningAuthorizeFilters](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs)
+* [InputFormatterExceptionPolicy](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs)
+
+[!code-csharp[Main](startup/sampleCompatibility/Startup2.cs?name=snippet1)]
+
+For apps that encounter breaking changes, using the appropriate compatibility switches:
+
+* Allows you to use the latest release and opt out of specific breaking changes.
+* Gives you time to update your app so it works with the latest changes.
+
+The [MvcOptions](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs) class source comments have a good explanation of what changed and why the changes are an improvement for most users.
+
+The comments in the [CompatibilitySwitch](https://github.com/aspnet/Mvc/blob/747420e5aa7cc2c7834cfb9731510286ded6fc03/src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/CompatibilitySwitch.cs#L9) provide more information on the breaking changes and instructions on how to implement a behavior change and corresponding compatibility switch.
+
+At some future date, there will be an [ASP.NET Core 3.0 version](https://github.com/aspnet/Home/wiki/Roadmap). Applications targeting ASP.NET Core 3.0 will not be able to opt-out of breaking changes introduced in ASP.NET Core 2.x. We feel these are positive changes benefitting nearly all users. By introducing these changes now, most apps can benefit now, and the others will have time to update their applications.
+
```

Comment:
I think it could be more positive to say that *old behaviors supported by compatibility switches will be removed in 3.0*